tools/debug-ui/src/BlocksView.tsx

```diff
@@ -0,0 +1,188 @@
+import './BlocksView.scss';
+import { useQuery } from 'react-query';
+import { fetchChainProcessingStatus, BlockProcessingStatus, ChunkProcessingStatus } from './api';
+
+type BlocksViewProps = {
+    addr: string;
+};
+
+function prettyTime(datetime: string) {
+    const time = new Date(Date.parse(datetime));
+    return (
+        time.getUTCHours() +
+        ':' +
+        String(time.getUTCMinutes()).padStart(2, '0') +
+        ':' +
+        String(time.getUTCSeconds()).padStart(2, '0') +
+        '.' +
+        String(time.getUTCMilliseconds()).padStart(3, '0')
+    );
+}
+
+function printTimeInMs(time: number | null) {
+    if (time == null) {
+        return 'N/A';
+    } else {
+        return time + 'ms';
+    }
+}
+
+function printDuration(start: string, end: string): string {
+    const duration = Date.parse(end) - Date.parse(start);
+    if (duration > 0) {
+        return '+' + duration + 'ms';
+    } else {
+        return duration + 'ms';
+    }
+}
+
+function getChunkStatusSymbol(chunkStatus: ChunkProcessingStatus): string {
+    switch (chunkStatus) {
+        case 'Completed':
+            return '✔';
+        case 'Requested':
+            return '⬇';
+        case 'NeedToRequest':
+            return '.';
+        default:
+            return '';
+    }
+}
+
+function printBlockStatus(blockStatus: BlockProcessingStatus): string {
+    switch (blockStatus) {
+        case 'Orphan':
+            return 'Orphan';
+        case 'WaitingForChunks':
+            return 'WaitingForChunks';
+        case 'InProcessing':
+            return 'InProcessing';
+        case 'Accepted':
+            return 'Accepted';
+        case 'Unknown':
+            return 'Unknown';
+        case 'HeightProcessed':
+            return 'HeightProcessed';
+        case 'TooManyProcessingBlocks':
+            return 'TooManyProcessingBlocks';
+        case 'Error':
+            return 'Error';
+        default:
+            return '';
+    }
```

Comment:
Is there a way to convert BlockProcessingStatus to string without having to do this?

Reply:
Thanks, the code provided doesnt compile, I did this instead (using valueOf and toString functions) and it does, but not 100%  sure if it's right

```
function printBlockStatus(blockStatus: BlockProcessingStatus): string {
    if ('Error' in blockStatus.valueOf) {
        return `Error: ${blockStatus.valueOf.Error}`;
      }
      if ('DroppedReason' in blockStatus.valueOf) {
        return `Dropped: ${blockStatus.valueOf.DroppedReason}`;
      }
      return blockStatus.toString();
}
```

Reply:
Fixed in 3rd commit as discussed offline, thanks for helpful explanation and the corrected code!